### ISSUE TITLE
Can you provide an example on how to use it?

### ISSUE THREAD

**Issue by cavalleydude** [edopelawi/CascadingTableDelegate#13]
I've read through the code, I don't understand how to use this in my project. Can you provide a step by step "how to use" section in the readme?

**Comment by edopelawi** [comment]
Hi @cavalleydude, thanks for asking!

Hmm... Sure, I'll try to write "easy start" section in the README. Meanwhile, would you take a look on the sample project?

**Comment by cavalleydude** [comment]
Yes, I ran the example app and have read all of the code. I think what I need to do is, format a tableView that shows the dataModel items, then on click I pass the model to the cascadingTableDelegate. I'm just trying to see a full set of data, where the I have an array of dataModels visually show them in one tableView, then show the detail on click with the DestinationViewController. Does that sound about right?

**Comment by cavalleydude** [comment]
To add more information to my comment above, I think the DestinationViewController is the Detail view of a Master table. So, if I implement another regular UITableViewController and use it to show the summary (master) row data, then clicking any row would open the detail DestinationViewController. That's how I'm thinking of implementing in my project.

**Comment by edopelawi** [comment]
Hmm. Let me run through your question, I need to make sure we're on the same page:

>  I think what I need to do is, format a tableView that shows the dataModel items, then on click I pass the model to the cascadingTableDelegate. I'm just trying to see a full set of data, where the I have an array of dataModels visually show them in one tableView, then show the detail on click with the DestinationViewController.

So, you're planning to make a TableView that has one type of cells in the first UIViewController, and pass the data of the selected cell (dataModel) to DestinationViewController. And, you're going to use CascadingTableDelegate to display a detail page with a UITableView with different type of cells, no? If that's your use case, then yes, you're doing it right :)

**Comment by edopelawi** [comment]
Please remember though, you'll need to do some coding on your side. One of the example might be creating an NSObject that conforms UITableViewDelegate and UITableViewDataSource to handle the "header" section of your detail page, and another to handle the "detail" section that will be shown using several cells of similar type. 

In above example, you will need to set a CascadingRootTableDelegate in your DestinationViewController, set it as your UITableView's dataSource and delegate, and pass the created "section delegates" as its childDelegates.

Do my explanation helps?

**Comment by cavalleydude** [comment]
Okay thank you. I will try and let you know.
